### ISSUE TITLE
Errors and/or unclear statements in Time Preference Fallacy related to interest rates

### ISSUE THREAD

**Issue by fresheneesz** [libbitcoin/libbitcoin-system#1357]
https://github.com/libbitcoin/libbitcoin-system/wiki/Time-Preference-Fallacy

> A higher interest rate implies a greater willingness for a person of a given time preference to lend.

I think what this is saying is that for someone of a given time preference, that person would be more willing to lend at a higher interest rate than a lower one (for equal amounts of risk taken on). This matches with the very next sentence that correcly says that falling time preference leads to falling interest rates. However, it reads as if you're saying that interest rates go up when the time preference of lenders go up. I suggest rewording.

> Zero time preference implies no consumption of what is produced.

This is not correct. What it instead implies is that the economic actor in question will choose the outcome that provides them with the highest value at the end of time. Very often that will be near-term actions you might consider "consumption". Let's take for example the simplified model of a robot economic actor. That robot has needs, say electricity. It needs to consume electricity in order to continue working and producing things. So even with 0 time preference, the right course of action is not to never consume electricity, instead the right course of action is to consume it periodically to keep it going. Humans are just more complex actors than that. Even with 0 time preference, a human will consume food and other goods and services to keep them physically and mentally healthy. Doing that does not imply a higher than 0 time preference. I think the key takeaway is that time preference involves "preference" when choosing among available options, but does not imply anything specific about consumption without considering those available options. 

For infinite time preference leading to no production, I can't find reasoning to argue against that, other than that time preference can be different for different goods. But if someone's time preference were truly infinite for all goods, I think I agree, nothing would be produced.

> Savings is the source of all investment, but only actual investment expresses time preference

This seems to be an invalid premise. Since many treat bitcoin as an investment they expect to rise in value, increasing your hoard does indeed imply a lower time preference. However, increasing a hoard of dollars instead, would not imply a lower time preference because its a bad store of value. At some point on the way up towards "full hoarding" the market value of bitcoin would exceed its true value and hoarding more would then reverse to being a higher time preference behavior, rather than 0 time preference. Even before it reaches that point, the upside potential of hoarding more would reduce below the average ROI of other investment opportunities in the market. Putting your money towards the highest return investment on a longer time horizon does imply higher time preference, regardless of whether than investment "produces" something or is just a simple store of value. Therefore, I don't think you can validly imply the contradictions you claim in that write up.

**Comment by evoskuil** [comment]
> I think what this is saying is that for someone of a given time preference, that person would be more willing to lend at a higher interest rate than a lower one

No, this is true for everyone. Everyone prefers to sell at a higher price and buy at a lower price.

**Comment by evoskuil** [comment]
>> Zero time preference implies no consumption of what is produced.

> This is not correct. What it instead implies is that the economic actor in question will choose the outcome that provides them with the highest value…

No, that is also a case of people prefer more value over less. That is not related to TP.

**Comment by evoskuil** [comment]
Sorry, there is just too much wrong with this to go on.

**Comment by fresheneesz** [comment]
> No, this is true for everyone always. Everyone prefers to sell at a higher price and buy at a lower price.

I certainly agree. However if you'll reread the quote from the page in question, you'll remember it literally says "for a person of a given time preference", so I was only reitterating that in my rephrasing of it in terms of what I understood it to say as I read it. The point is, the sentence I was commenting on is not clearly worded and is misleading as written.

> No, that is also a case of all people always prefer more value (what they prefer) over less. That is not related to TP.

I respectfully disagree. A person with a higher than 0 time-preference will most assurely not be optimizing for maximum total wealth at the end of time, regardless of the true fact that everyone prefers more (utility) over less. That person instead will make decisions that brings economic utility to them nearer in time to them, at the expense of their wealth at the end of time. This is basically the fundamental truth about time preference, so its very odd to me that you say its not related. 

I'm sorry to hear that you think this isn't worth discussing. But since you didn't really provide much reasoning behind your responses to my arguments, I don't know why you're making those assertions.

**Comment by evoskuil** [comment]
>> A higher interest rate implies a greater willingness for a person of a given time preference to lend.
 > it reads as if you're saying that interest rates go up when the time preference of lenders go up
It is true that interest rates are higher ("go up") when the TP of people ("lenders") is higher - higher TP implies greater demand for return on lending. Yet that is clearly not what the line says.

A "person of a given time preference" has a "greater willingness to lend" in the case of a "higher interest rate" (i.e. higher than otherwise). Now, all people have a "given time preference" - this phrase is just to clarify that the person's time preference is fixed in the scenario, not changing. In other words, people prefer to lend at higher rates. This statement is to refute the common error that time preference changes as a function of interest rates. Time preference *determines* interest rates, and therefore such claims are circular. I see no problem with the statement and, while I don't see how you obtain your interpretation of it, that interpretation is also a true statement.

**Comment by evoskuil** [comment]
> all people always prefer more value (what they prefer) over less.

This statement has no dependency on time preference, that is self-evident. You describing a time preference scenario does not change that fact. It is time preference depends on this statement.

**Comment by evoskuil** [comment]
>>Zero time preference implies no consumption of what is produced.
>This is not correct. What it instead implies is that the economic actor in question will choose the outcome that provides them with the highest value…
It is correct. A person with zero time preference would lend everything at any interest rate. The person would have no property and would perish of deprivation. If all people have zero time preference, they all perish. They must exhibit some non-zero demand for consumption. I don't see this as a particularly challenging statement. It is also the case that infinite time preference implies the end of humanity, as nothing could be produced and eventually everything hoarded would be consumed (as you agreed in your first post).

Your argument that it wasn't correct was a axiomatic platitude about people doing what they prefer. It's not a refutation in any way, just a truism.

**Comment by evoskuil** [comment]
>> Savings is the source of all investment, but only actual investment expresses time preference

> This seems to be an invalid premise. Since many treat bitcoin as an investment they expect to rise in value, increasing your hoard does indeed imply a lower time preference.

It is not. Gambling doesn't become investment simply because someone expects to win. Investment economically is production. Hoarding is not producing, and a greater tendency to hoard implies a higher time preference. This is why the interest rate is a direct reflection of TP - it's about investment, investment is about production, and interest is the return on production. TP has nothing to do with speculation (gambling in price changes). That is the whole point of the chapter, many Bitcoiners have been conned by false economics.

People must *produce* to survive. If everyone hoards everything, interest rates are infinite because there is nothing being produced (and there is no actual interest, because there is no return on production). Higher hoarding implies higher rates. Higher rates reflect higher time preference. Hoarding your coins is higher time preference vs. lending them to production, which is lower.

**Comment by evoskuil** [comment]
> I'm sorry to hear that you think this isn't worth discussing.
This is not a discussion forum. If you have questions I'm happy to discuss, but this is a software issue board.

> But since you didn't really provide much reasoning behind your responses to my arguments
Bullshit, I answered with valid reasons. Verbosity is not what makes an argument.

**Comment by fresheneesz** [comment]
> I see no problem with the statement

Fair enough, my goal was only to point out my confusion around it with the thought that others might read it with similar confusion. 

> A person with zero time preference would lend everything at any interest rate. The person would have no property and would perish of deprivation.

If that's your conclusion, than I'm using a different definition of time-preference than you. I think this is where all of our disagreement must lie.

> This is not a discussion forum

Thanks for your more in depth responses. Perhaps I'll find a better forum for this discussion at some point.

**Comment by evoskuil** [comment]
Like many bitcoiners you may be using the psychology term, associated with the marshmallow experiment (deferred consumption). That conflation was put forth by a popular book. This is not related to the economics concept and has led to a lot of contusion.
